simulation/types/txbuilder.go

```diff
@@ -57,3 +62,62 @@ func (sim *SimCtx) deliverTx(tx sdk.Tx, msg sdk.Msg, msgName string) (simulation
 
 	return simulation.NewOperationMsg(msg, true, "", nil), nil, nil
 }
+
+// GenTx generates a signed mock transaction.
+// TODO: Surely theres proper API's in the SDK for this?
+// (This was copied from SDK simapp, and deleted the egregiously non-deterministic memo handling)
+func genTx(gen client.TxConfig, msgs []sdk.Msg, feeAmt sdk.Coins, gas uint64, chainID string, accNums, accSeqs []uint64, priv ...cryptotypes.PrivKey) (sdk.Tx, error) {
```

Comment:
Added to fix the memo being non-deterministic in the SDK simulator. Theres got to be more standard API's so we don't import all this complexity right here though.

Reply:
There might be methods/APIs in `client/tx` package that help here.